src/hypothesis/strategies.py

```diff
@@ -1287,6 +1361,27 @@ def __not_a_first_class_strategy(self, name):
             ) % (name,))
     return DataStrategy()
 
+
+def register_type_strategy(custom_type, strategy):
+    """Add an entry to the global type-to-strategy lookup.
+
+    This lookup is used in :func:`~hypothesis.strategies.builds` and
+    :func:`@given <hypothesis.given>`.
+
+    :func:`~hypothesis.strategies.builds` will be used automatically for
+    classes with type annotations, so you only need to register a strategy
```

Comment:
Do you mean type annotations on their constructor? If so I think it should say that explicitly. If not, I don't know what this means so it needs clarification.